### PR TITLE
Legacy fiducial and differential xs measurement dev PR

### PR DESCRIPTION
**Pull request by threiten**

### CODE REVIEW

File: DataFormats/src/classes_def_94X.xml [hunk 1]
```diff
@@ -0,0 +1,392 @@
+<lcgdict>
```
Comment (simonepigazzini) [inline DataFormats/src/classes_def_94X.xml]: This file went in by mistake, should be removed

File: MetaData/data/cross_sections.json [hunk 2]
```diff
@@ -125,9 +125,12 @@
     "DoubleMuon"                                                    : { "xs" : 0.0, "itype" :0                       },
     "Higgs_scalar" : {"xs":1, "br":1, "kf":1.0, "itype":-1125},
 
-    "GluGluHToGG_M-125_13TeV_powheg_pythia8"                        : { "xs" : 44.14,   "br" : 2.27e-3, "kf" : 1.06, "itype":-125000 },
+	"GluGluHToGG_M-120_13TeV_powheg_pythia8"                        : { "xs" : 47.38,   "br" : 2.22e-3, "kf" : 1.06, "itype":-120000 },
```
Comment (simonepigazzini) [inline MetaData/data/cross_sections.json:128]: @edjtscott is it possible that these xs were missing?

File: MetaData/python/JobConfig.py [hunk 3]
```diff
@@ -5,12 +5,15 @@
 
 import commands
 
+<<<<<<< HEAD
```
Comment (simonepigazzini) [inline MetaData/python/JobConfig.py]: Houston, we have a problem :)

File: MetaData/work/campaigns/RunIIFall17-3_2_0_part1.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (simonepigazzini) [inline MetaData/work/campaigns/RunIIFall17-3_2_0_part1.json]: obsolete file, please remove

File: MetaData/work/campaigns/RunIIFall17-3_2_0_part2.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (simonepigazzini) [inline MetaData/work/campaigns/RunIIFall17-3_2_0_part2.json]: same

File: MicroAOD/interface/CutBasedElectronObjectSelector.h [hunk 6]
```diff
@@ -1,13 +1,62 @@
 #ifndef flashgg_CutBasedElectronObjectSelector_h
```
Comment (simonepigazzini) [inline MicroAOD/interface/CutBasedElectronObjectSelector.h:1]: Is this code used for the differential analysis or for the T&P?

File: Systematics/plugins/PhotonSigEoverEShift.cc [hunk 7]
```diff
@@ -57,7 +57,7 @@ namespace flashgg {
                     std::cout << beforeSigEoE << " ";
                     std::cout << std::endl;
                 }
-                y.shiftSigmaEOverEValueBy( shift ); 
+                y.shiftSigmaEOverEValueBy( shift, 0.00 );
```
Comment (simonepigazzini) [inline Systematics/plugins/PhotonSigEoverEShift.cc:60]: Is this a bugfix or a compatibility issue?

File: Systematics/test/sig_jobs_vrt_fiducial_IA_trees_compCand.sh [hunk 8]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (simonepigazzini) [inline Systematics/test/sig_jobs_vrt_fiducial_IA_trees_compCand.sh]: these files (if needed at all) should go into a subdirectory (the same that is done for the STXS)

File: Systematics/test/splitTree.py [hunk 9]
```diff
@@ -0,0 +1,150 @@
+import tree2dataset as t2d
```
Comment (simonepigazzini) [inline Systematics/test/splitTree.py]: these script too should go into a subdirectory, actually if you put them under Systematics/scripts they will be available system wide trhough $PATH

File: Taggers/test/MVATraining/checkJsons.py [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import sys
+import os
+
+sys.path.append('/cvmfs/cms.cern.ch/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_0_26/bin/slc6_amd64_gcc530/')
```
Comment (simonepigazzini) [inline Taggers/test/MVATraining/checkJsons.py]: ???

**Review by simonepigazzini — CHANGES_REQUESTED** [review]

**Comment by simonepigazzini** [comment]
@edjtscott, @jonathon-langford , I know you are very busy with the STXS getting to the end of it, but if you can have a look at this running few samples to check nothing breaks on your side it would really help. I think the main sources of troubles could be located in the Dumpers, for the rest the changes are minor (changes in the taggers are usually just indentation or newlines that went in somehow).

File: MicroAOD/interface/CutBasedElectronObjectSelector.h (hunk 6 above)
Comment (threiten) [reply MicroAOD/interface/CutBasedElectronObjectSelector.h:1]: Comes from here commit https://github.com/threiten/flashgg/commit/59f86a37aea8a194bfc2500f44fdd9aab647a999#diff-5d1343eef6f9b26821aff25349a9c9f3
So used for differentials.

File: Taggers/test/MVATraining/checkJsons.py (hunk 10 above)
Comment (threiten) [reply Taggers/test/MVATraining/checkJsons.py]: The whole MVATraining directory is ancient. I think it can go completely

File: MicroAOD/interface/CutBasedElectronObjectSelector.h (hunk 6 above)
Comment (simonepigazzini) [reply MicroAOD/interface/CutBasedElectronObjectSelector.h:1]: ok

File: Taggers/test/MVATraining/checkJsons.py (hunk 10 above)
Comment (simonepigazzini) [reply Taggers/test/MVATraining/checkJsons.py]: I agree

Comment (threiten) [reply Taggers/test/MVATraining/checkJsons.py]: Removed it from my branch

File: Systematics/plugins/PhotonSigEoverEShift.cc (hunk 7 above)
Comment (threiten) [reply Systematics/plugins/PhotonSigEoverEShift.cc:60]: Compatibility, the function in Photon.cc is changed. https://github.com/threiten/flashgg/commit/4d6f19d5e63b06850fb2c0a1339078562f658e54

Comment (simonepigazzini) [reply Systematics/plugins/PhotonSigEoverEShift.cc:60]: ok thanks

**Review by simonepigazzini — APPROVED** [review]

**Comment by simonepigazzini** [comment]
@edjtscott , @jonathon-langford we would really like to merge this, it has been pending since 2016 :). Let me know if you have any objection.

**Comment by edjtscott** [comment]
sure I will look through the code quickly & start testing now - sorry for the delay

File: Taggers/python/flashggUpdatedIdMVADiPhotons_cfi.py [hunk 11]
```diff
@@ -29,11 +29,11 @@
                                               ## Shower shape correction (5x5)
                                               do5x5correction          = cms.bool(False), ## Turn this off to remove 5x5 shower shape corrections
                                               # correctionFile           = cms.FileInPath("flashgg/MicroAOD/data/transformation5x5_Legacy2016_v1.root"), ## for Legacy2016 
-                                              correctionFile           = cms.FileInPath("flashgg/MicroAOD/data/transformation5x5_ReReco2017_v2.root"),  ## for Rereco2017
```
Comment (edjtscott) [inline Taggers/python/flashggUpdatedIdMVADiPhotons_cfi.py:32]: just to check: this is no longer used in the standard sequences, right? So these changes are not expected to affect the results of any other analyses.

File: DataFormats/src/Jet.cc [hunk 12]
```diff
@@ -67,6 +73,117 @@ float Jet::betaStar( const edm::Ptr<DiPhotonCandidate> dipho ) const
     return betaStar( dipho->vtx() );
 }
 
+
+bool Jet::passesJetPuId( JetPuIdLevel level ) const
```
Comment (edjtscott) [inline DataFormats/src/Jet.cc:77]: this development is very welcome but can you please check it follows the latest recommendations? We updated the treatment in e.g. the VBF MVA [producer](https://github.com/cms-analysis/flashgg/blob/dev_legacy_runII/Taggers/plugins/VBFMVAProducer.cc#L253-L265) and [config](https://github.com/cms-analysis/flashgg/blob/dev_legacy_runII/Taggers/python/flashggVBFMVA_cff.py#L5-L12) a while ago to conform to the [recommendations](https://twiki.cern.ch/twiki/bin/viewauth/CMS/PileupJetID). The main one is we are no longer meant to apply any cut above a jet pT of 50 GeV.

File: Taggers/interface/CategoryDumper.h [hunk 13]
```diff
@@ -162,6 +163,7 @@ namespace flashgg {
         std::vector<std::shared_ptr<wrapped_stepwise_functor_type> > stepwise_functors_;
         bool hbooked_;
         bool binnedOnly_;
+        bool unbinnedSystematics_;
```
Comment (edjtscott) [inline Taggers/interface/CategoryDumper.h:166]: out of curiosity, what is the motivation for these changes? Are any of the default behaviours changed?

File: Taggers/interface/CollectionDumper.h [hunk 14]
```diff
@@ -125,7 +127,8 @@ namespace flashgg {
         bool dumpWorkspace_;
         std::string workspaceName_;
         bool dumpHistos_, dumpGlobalVariables_;
-
+        // bool dumpNNLOPSweight_;
```
Comment (edjtscott) [inline Taggers/interface/CollectionDumper.h:130]: thanks for cleaning this up!

File: Systematics/test/workspaceStd.py [hunk 15]
```diff
@@ -670,7 +769,66 @@
...
+#                          process.flashggDiPhotonSystematics*
+#                          process.flashggMetSystematics*
+#                          process.flashggMuonSystematics*process.flashggElectronSystematics*
+#                          (process.flashggUnpackedJets*process.jetSystematicsSequence)*
+#                          (process.flashggTagSequence*process.systematicsTagSequences)*
+#                          process.flashggSystTagMerger*
+#                          process.penultimateFilter*
+#                          process.finalFilter*
+#                          process.tagsDumper)
+if not customize.processId=="Data" :
```
Comment (edjtscott) [inline Systematics/test/workspaceStd.py]: after some testing it looks like the logic has been changed here - presumably unintended? This file also needs cleanup in general, so please have a look (did you mean to replace with fiducialStd?)

Comment (threiten) [reply Systematics/test/workspaceStd.py]: In principle everything for the fiducial went from workspaceStd to fiducialStd. I will remove everything from here. For workspaceStd, the idea was to take whatever is there in the master. Is missed this bc there was no conflict here. Sorry for that.

File: Taggers/python/flashggUpdatedIdMVADiPhotons_cfi.py (hunk 11 above)
Comment (threiten) [reply Taggers/python/flashggUpdatedIdMVADiPhotons_cfi.py:32]: This was replaced entirely by the differential corrections. Actually its a good idea to get rid of this whole file.

File: DataFormats/src/Jet.cc (hunk 12 above)
Comment (threiten) [reply DataFormats/src/Jet.cc:77]: Thanks for pointing this out. I will adjust the cuts. Am I right that the WPs mixed and forward_* are gone ?

File: Systematics/test/workspaceStd.py [hunk 16]
```diff
@@ -138,16 +139,34 @@
                            VarParsing.VarParsing.varType.bool,
                            'doFiducial'
                            )
+customize.options.register('doBJetsAndMET',
```
Comment (simonepigazzini) [inline Systematics/test/workspaceStd.py]: These options are no longer needed in workspaceStd. I'm not sure if @threiten already removed these but they are still showing up in github, or if they are really still there

File: DataFormats/src/Jet.cc (hunk 12 above)
Comment (edjtscott) [reply DataFormats/src/Jet.cc:77]: yes looks like it, and I am pretty sure nobody is using them within flashgg either

File: Taggers/interface/CategoryDumper.h (hunk 13 above)
Comment (threiten) [reply Taggers/interface/CategoryDumper.h:166]: I think this is needed to dump some variables from extra objects attached to the diphoton. For the details, @vtavolar could you comment further please ?

**Comment by edjtscott** [comment]
FYI I tested again with workspaceStd fixed and all seems to run OK, so once the few remaining changes are made it's fine to merge from my side

**Comment by simonepigazzini** [comment]
Thanks guys, merging now (finally after ~4 years :) )
